smartsim/_core/control/interval.py

```diff
@@ -0,0 +1,112 @@
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+
+import time
+import typing as t
+
+Seconds = t.NewType("Seconds", float)
+
+
+class SynchronousTimeInterval:
+    """A utility class to represent and synchronously block the execution of a
+    thread for an interval of time.
+    """
+
+    def __init__(self, delta: float | None) -> None:
+        """Initialize a new `SynchronousTimeInterval` interval
+
+        :param delta: The difference in time the interval represents. If
+            `None`, the interval will represent an infinite amount of time in
+            seconds.
+        :raises ValueError: The `delta` is negative and `strict` is `True`
+        """
+        if delta is not None and delta < 0:
+            raise ValueError("Timeout value cannot be less than 0")
+        if delta is None or delta < 0:
+            delta = float("inf")
+        self._delta = Seconds(delta)
+        """The amount of time, in seconds, the interval spans."""
+        self._start = time.perf_counter()
+        """The time of the creation of the interval"""
+
+    @property
+    def delta(self) -> Seconds:
+        """The difference in time the interval represents
+
+        :returns: The difference in time the interval represents
+        """
+        return self._delta
+
+    @property
+    def elapsed(self) -> Seconds:
+        """The amount of time that has passed since the interval was created
+
+        :returns: The amount of time that has passed since the interval was
+            created
+        """
+        return Seconds(time.perf_counter() - self._start)
+
+    @property
+    def remaining(self) -> Seconds:
+        """The amount of time remaining in the interval
+
+        :returns: The amount of time remaining in the interval
+        """
+        return Seconds(max(self.delta - self.elapsed, 0))
+
+    @property
+    def expired(self) -> bool:
+        """The amount of time remaining in interval
```

Comment:
This has the same description as `def remaining`. Change to appropriate description

Reply:
🤦